build/nlu-regression.js

```diff
@@ -0,0 +1,218 @@
+      res.on('error', function(err) {
+        reject(err)
+      })
+      res.on('end', function() {
+        resolve(receivedData)
+      })
+    })
+
+    // post the data
+    post_req.write(content)
+    post_req.end()
+  })
+}
+
+async function get(path, headers) {
+  var post_options = {
+    host: '127.0.0.1',
+    port: '3000',
+    path,
+    method: 'GET',
+    headers
+  }
+
+  return new Promise(function(resolve, reject) {
+    var get_req = http.request(post_options, function(res) {
+      res.setEncoding('utf8')
+
+      let receivedData = ''
+      res.on('data', function(chunk) {
+        receivedData += chunk
+      })
+      res.on('error', function(err) {
+        reject(err)
+      })
+      res.on('end', function() {
+        resolve(receivedData)
+      })
+    })
+    get_req.end()
+  })
+}
+
+async function login() {
+  var data = 'email=admin&password=123456'
+  const rawLogin = await post('/api/v1/auth/login/basic/default', data, {
+    'Content-Type': 'application/x-www-form-urlencoded',
+    'Content-Length': Buffer.byteLength(data)
+  })
+  const login = JSON.parse(rawLogin)
+
+  const token = login.payload && login.payload.token
+  return login.statusCode ? undefined : token
+}
+
+async function signup() {
+  var data = 'email=admin&password=123456'
+  const rawLogin = await post('/api/v1/auth/register/basic/default', data, {
+    'Content-Type': 'application/x-www-form-urlencoded',
+    'Content-Length': Buffer.byteLength(data)
+  })
+  const login = JSON.parse(rawLogin)
+
+  const token = login.payload && login.payload.token
+  return login.statusCode ? undefined : token
+}
+
+async function createBot(botId, token) {
+  const newBot = JSON.stringify({
+    id: botId,
+    name: 'testy',
+    template: {
+      id: 'bp-nlu-regression-testing',
+      moduleId: 'nlu-testing'
+    },
+    category: undefined
+  })
+
+  await post('/api/v1/admin/bots', newBot, {
+    Authorization: `Bearer ${token}`,
+    'Content-Type': 'application/json',
+    'Content-Length': Buffer.byteLength(newBot),
+    'X-BP-Workspace': 'default'
+  })
+}
+
+async function waitForTraining(botId, token) {
+  return new Promise(function(resolve) {
+    var i = 0
+    console.log(`training...`)
+    const intervalId = setInterval(async () => {
+      const raw = await get(`/api/v1/bots/${botId}/mod/nlu/train`, {
+        Authorization: `Bearer ${token}`
+      })
+      const trainingStatus = JSON.parse(raw)
+      if (!trainingStatus.isTraining) {
+        clearInterval(intervalId)
+        resolve()
+      } else {
+        console.log(`training... ${2 * ++i}s`)
+      }
+    }, 2000)
+  })
+}
+
+function round(n, acc) {
+  var num = Math.pow(10, acc)
+  return Math.round(n * num) / num
+}
+
+async function runAllTests(botId, token) {
+  const baseNluTesting = `/api/v1/bots/${botId}/mod/nlu-testing`
+  const rawTests = await get(`${baseNluTesting}/tests`, {
+    Authorization: `Bearer ${token}`,
+    'Content-Type': 'application/json'
+  })
```

Comment:
Can you move this to something like "axiosConfig" to avoid repeating it in every method?